aten/src/ATen/native/quantized/cpu/qnnpack_utils.h

```diff
@@ -84,3 +89,58 @@ inline std::pair<uint8_t, uint8_t> activationLimits(
   }
 }
 #endif
+
+namespace {
+std::vector<float> generate_requantization_scales(
+    const at::Tensor& weight_scales,
+    const float input_scale,
+    const float output_scale) {
+  auto num_output_channels = weight_scales.numel();
+  float* weight_scales_data = weight_scales.data_ptr<float>();
+  std::vector<float> requant_scales(num_output_channels, 1.f);
+  for (int i = 0; i < num_output_channels; ++i) {
+    requant_scales[i] = weight_scales_data[i] * input_scale / output_scale;
```

Comment:
Could we check requant_scale values here rather than in the run function?

Reply:
We check the constraints inside qnnpack. Is that what you mean? Not sure what you mean by inside run function. This function is called from run function and values are checked during conv_param_t creation.

Reply:
I will try to move the check here if it seems more natural.